### PR TITLE
`Programming exercises`: Allow editing submission policy on exercise creation and import

### PR DESCRIPTION
**Pull request by basak-akan**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [ ] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Editing submission policy should be possible while creating and importing new programming exercises. However, the submission policy cannot be edited on the edit programming exercise page, it is only possible over the grading page. 
During the import programming exercise;
- If an exercise has a predefined submission policy, the import operation fails even if there are no changes made to the submission policy (the UI shows no operations, and there's a '500 unable to import programming exercise' error in the console). 
- If the imported exercise doesn't have a submission policy, an internal server error occurs when the submission policy is set.

### Description

Submission policy editing is enabled for creating and importing exercises (enabled for not in edit mode).
An indicator that the policy can be changed on the grading page with a link is added.
Import programming exercise submission policy issues fixed.

### Steps for Testing

- 1 Instructor

1. Create/Import a programming exercise 
2. Verify submission policy editing is enabled
3. Set the submission policy and save
4. Navigate to edit programming exercise
5. Verify submission policy is disabled
6. Verify an indicator with a link is shown
7. Verify link navigates to the grading page of the exercise edit policy tab

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Exam
- 1 Exercise Group

1. Create/Import a programming exercise in the exercise group
2. Verify submission policy editing is enabled
3. Set the submission policy and save
4. Navigate to edit programming exercise
5. Verify submission policy is disabled
6. Verify an indicator with a link is shown
7. Verify link navigates to the grading page of the exercise edit policy tab

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test 1
- [x] Test 2

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ProgrammingExerciseImportBasicService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5256/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseImportBasicService.html) | 100% | ✅ |
| [ProgrammingExerciseExportImportResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5256/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ProgrammingExerciseExportImportResource.html) | 96% | ✅ |

### Screenshots

Create Programming Exercise (Submission policy is editable):
image
Edit Programming Exercise (Submission policy is not enabled. A banner with a link is shown to user)
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java [hunk 1]
```diff
@@ -116,6 +116,7 @@ public ProgrammingExercise importProgrammingExerciseBasis(final ProgrammingExerc
         final Map<Long, Long> newTestCaseIdByOldId = importTestCases(templateExercise, importedExercise);
         final Map<Long, Long> newTaskIdByOldId = importTasks(templateExercise, importedExercise, newTestCaseIdByOldId);
         updateTaskExerciseHintReferences(templateExercise, importedExercise, newTaskIdByOldId, newHintIdByOldId);
+        importSubmissionPolicy(importedExercise);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java:119]: Why is the order relevant here?
Maybe leave a comment in the code explaining it

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 2]
```diff
@@ -507,6 +507,7 @@
                 "submissionLimitTitle": "Abgabelimit",
                 "submissionLimitDescription": "Die Anzahl von erlaubten Abgaben, bis das System die ausgewählte Abgaberichtlinie durchsetzt.",
                 "editInGradingInformation": "Das Abgabelimit kann nur in der Bewertungssicht der Programmieraufgabe bearbeitet und (de)aktiviert werden.",
+                "goToGradingToEditInformation": "Gehen Sie zur Seite Benotung, um die Einreichungsrichtlinien zu bearbeiten.",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/programmingExercise.json:510]:
```suggestion
                "goToGradingToEditInformation": "Gehe zur Bewertungsseite, um die Einreichungsrichtlinien zu bearbeiten.",
```

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 3]
```diff
@@ -502,6 +502,7 @@
                 "submissionLimitTitle": "Submission limit",
                 "submissionLimitDescription": "The number of submissions a participant can make before the system enforces the selected policy.",
                 "editInGradingInformation": "The submission policy can only be edited and toggled on the grading page of the programming exercise!",
+                "goToGradingToEditInformation": "Go to the Grading Page to edit Submission Policy.",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/programmingExercise.json:505]:
```suggestion
                "goToGradingToEditInformation": "Go to the grading page to edit submission policy.",
```
This is a proper sentence and not a heading

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 4]
```diff
@@ -507,6 +507,7 @@
                 "submissionLimitTitle": "Abgabelimit",
                 "submissionLimitDescription": "Die Anzahl von erlaubten Abgaben, bis das System die ausgewählte Abgaberichtlinie durchsetzt.",
                 "editInGradingInformation": "Das Abgabelimit kann nur in der Bewertungssicht der Programmieraufgabe bearbeitet und (de)aktiviert werden.",
+                "goToGradingToEditInformation": "Gehe zur Bewertungsseite, um die Einreichungsrichtlinien zu bearbeiten.",
```
Comment (MaximilianAnzinger) [inline src/main/webapp/i18n/de/programmingExercise.json:510]:
```suggestion
                "goToGradingToEditInformation": "Gehe zur Bewertungsseite, um die Abgaberichtlinie zu bearbeiten.",
```

**Review by MaximilianAnzinger — COMMENTED** [review]
Overall code quality looks good. Left one comment.

**Review by JohannesStoehr — DISMISSED** [review]
Approve Code

**Review by terlan98 — DISMISSED** [review]
Tested on TS5. Works as expected 👍 The submission policy is editable during the creation and import of an exercise. Later, it can be edited only from the grading page.

I think it would be better to put the new hintbox containing the link to the grading page before the Submission Limit and Exceeding Submission Limit Penalty
Screenshot

Also tested the internal server errors mentioned in the description. The 1st error is solved by this PR. _However, I could not reproduce the 2nd error on `develop` and thus can't really verify whether it exists._

**Review by laadvo — APPROVED** [review]
Tested on ts1, the submission policy can be edited when an exercise gets created and imported. No internal server errors appeared when importing

**Review by MaximilianAnzinger — DISMISSED** [review]
Code LGTM 👍

**Review by JohannesStoehr — APPROVED** [review]
Reapprove code

**Review by terlan98 — APPROVED** [review]
Re-tested on TS5. Still works 👍

**Review by rriyaldhi — APPROVED** [review]
Code lgtm 👍

**Review by bassner — APPROVED** [review]

**Pull request merged**